frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/bc.ts

```diff
@@ -97,28 +157,71 @@ const setBackingStores = (tiers: Tier[]) => {
       expect($items).toHaveLength(2);
     });
   }
-  selectBackingStore(1, tests.pop());
+  selectStoreFromTable(1, tests.pop());
   if (TierCountMap[tiers[0]] > 1) {
-    selectBackingStore(1, tests.pop());
+    selectStoreFromTable(1, tests.pop());
   }
   // Select tier 2 Backing Stores
   if (tiers.length > 1) {
-    selectBackingStore(2, tests.pop());
+    selectStoreFromTable(2, tests.pop());
     if (TierCountMap[tiers[1]] > 1) {
-      selectBackingStore(2, tests.pop());
+      selectStoreFromTable(2, tests.pop());
     }
   }
 };
 
-export const createBC = (tiers: Tier[]) => {
-  setGeneralData();
-  cy.contains('Next').click();
-  // Placement Policy Page
-  setPlacementPolicy(tiers);
+const selectItemFromStoreDropdown = (name: string, type: StoreType) => {
+  cy.byTestID(`${type === StoreType.NamespaceStore ? 'nns' : 'nbs'}-dropdown-toggle`).click();
+  cy.byTestID(`${name}-dropdown-item`).click();
+};
+
+const NamespaceBucketClassFlows = {
+  [NamespacePolicyType.SINGLE]: (config: NamespaceBucketClassConfig) => {
+    selectItemFromStoreDropdown(config.resources[0], StoreType.NamespaceStore);
+  },
+  [NamespacePolicyType.MULTI]: (config: NamespaceBucketClassConfig) => {
+    selectStoreFromTable(1, config.resources[0]);
+    selectStoreFromTable(1, config.resources[1]);
+    selectItemFromStoreDropdown(config.resources[0], StoreType.NamespaceStore);
+  },
+  [NamespacePolicyType.CACHE]: (config: NamespaceBucketClassConfig) => {
+    selectItemFromStoreDropdown(config.resources[0], StoreType.NamespaceStore);
+    selectItemFromStoreDropdown(config.testBackingStore, StoreType.BackingStore);
+    cy.byTestID('time-to-live-input').type('2');
+  },
+};
+
+export const getBucketClassConfig = (
+  resources: string[],
+  type: BucketClassType,
+): BucketClassConfig =>
+  type === BucketClassType.STANDARD
+    ? new StandardBucketClassConfig(resources, type)
+    : new NamespaceBucketClassConfig(resources, type);
```

Comment:
I think we can avoid this and call classes explicitly. It seems an over specification.
It will be easier to follow.
```suggestion
```

Reply:
alright... We can do it either ways.